Archipelago.h

```diff
@@ -272,11 +273,13 @@ std::vector<AP_Gift> AP_CheckGifts();
 // IDs and Sender Info are set by the library. The values set will be ignored
 AP_RequestStatus AP_SendGift(AP_Gift gift);
 
-// Accept a gift from the Giftbox, and writes it into a struct in the second parameter. ONLY THIS DATA IS "REAL", DO NOT REUSE DATA FROM AP_CheckGifts()
-AP_RequestStatus AP_AcceptGift(std::string id, AP_Gift* gift);
```

Comment:
Gift acceptance should still stay as it was, meaning using a gift is only valid once returned from acceptgift. This is to ensure consistency with the server

Reply:
even after our brief discussion on discord, i still fail to see how this adds any level of consistence, it does add a lot of extra complication to the code, i guess the only case where the data in AP_CheckGifts() different might be in the super edge case of partial gift acceptance, like where a Gift is 10 items but you only take out 7 or so and leave the rest for later or reject

Reply:
Point is to separate the query from an actual, usable item. With the gifting api, these are two different lookups, and could conflict. All of this is "in theory" though, and in practice makes no real difference. But, that is imo what the apcpp api then should do, while, in the internal implementation, it can be a cache lookup.

Reply:
I am not a fan of this added complexity that i am not going to use myzelf but since your verry pasionate about it i can see to bring it back

how would this work for bulk accept? i dont think we can do `std::map<std::string, AP_Gift>*` ?

Reply:
You can use a set of `AP_Gift`s, they have the ID embedded.

One thought: If an ID does not exist among the set, AP will still pop the other ones in order of the operations right? So this will probably be lossy on error. Wonder of there's a better way...

Reply:
Discussed in DM's we will leave the out variables out of Accept gifts, while they make sense from some design standpoints it isn't worth the extra complexity